src/ts/components/subtitleoverlay.ts

```diff
@@ -416,3 +418,82 @@ class ActiveSubtitleManager {
     this.activeSubtitleCueCount = 0;
   }
 }
+
+class SubtitleRegionContainerManager {
+  private subtitleRegionContainers: { [region: string]: SubtitleRegionContainer } = {};
+  private subtitleOverlay: SubtitleOverlay;
+
+  /**
+   * @param subtitleOverlay Reference to the subtitle overlay for adding and removing the containers.
+   */
+  constructor(subtitleOverlay: SubtitleOverlay) {
+    this.subtitleOverlay = subtitleOverlay;
+  }
+
+  /**
+   * Removes a subtitle label from a container.
+   * @param labelToRemove
+   */
+  public removeLabel(labelToRemove: SubtitleLabel): void {
+    for (let region in this.subtitleRegionContainers) {
+      this.subtitleRegionContainers[region].removeLabel(labelToRemove);
+    }
+  }
+
+  /**
+   * Removes all subtitle containers.
+   */
+  public clear(): void {
+    for (let region in this.subtitleRegionContainers) {
+      this.subtitleOverlay.removeComponent(this.subtitleRegionContainers[region]);
+    }
+
+    this.subtitleRegionContainers = {};
+  }
+
+  /**
+   * Creates and wraps a subtitle label into a container div based on the subtitle region.
+   * If the subtitle has positioning information it is added to the container.
+   * @param label The subtitle label to wrap
+   * @param event Subtitle cue event that contains the positoning information
+   */
+  public wrap(label: SubtitleLabel, event: SubtitleCueEvent): void {
+    const subtitleRegion = event.region || 'default';
+      if (!this.subtitleRegionContainers[subtitleRegion]) {
+      const regionContainer = new SubtitleRegionContainer({
+        cssClass: `subtitle-position-${subtitleRegion}`,
+      });
+
+      this.subtitleRegionContainers[subtitleRegion] = regionContainer;
+
+      if (event.regionStyle) {
+        regionContainer.getDomElement().attr('style', event.regionStyle + ';position:absolute');
+      } else {
+        // getDomElement needs to be called at least once to ensure the component exists
+        regionContainer.getDomElement();
+      }
+    }
+
+    this.subtitleRegionContainers[subtitleRegion].addComponent(label);
+
+    for (let region in this.subtitleRegionContainers) {
```

Comment:
`const region`

Reply:
It might be better to call the variable `regionName` (everywhere) to make it obvious that this is the name of the region and not the actual region (container).

Reply:
Also I think you can again do the `for ... of`